### PR TITLE
feat(registration): add lookup by wallet address

### PR DESCRIPTION
**Pull request by enddynayn**
Add ability to lookup registrations by wallet address.

### CODE REVIEW

File: src/core/contracts/registry.test.ts [hunk 1]
```diff
@@ -506,4 +508,111 @@ describe("registry", () => {
       });
     });
   });
+
+  describe("#getRegistratiosByPublicAddress", () => {
```
Comment (rosalinekarr) [inline src/core/contracts/registry.test.ts:512]:
```suggestion
  describe("#getRegistrationsByPublicAddress", () => {
```

**Review by rosalinekarr — APPROVED** [review]
LGTM, one typo

File: src/core/contracts/registry.ts [hunk 2]
```diff
@@ -231,3 +231,19 @@ const getLatestRegistryUpdatesFor = async (identityAddress: HexString): Promise<
...
 };
+
+/**
+ * getRegistrationsByWalletAddress() Resolves all registrations associated to an identity contract address
+ *
+ * @param walletAddress - The address to resolve the registrations for
+ * @returns A list of registrations that are associated to the address
+ */
+export const getRegistrationsByWalletAddress = async (walletAddress: EthereumAddress): Promise<Registration[]> => {
+  const identitieAddresses: EthereumAddress[] = await getDelegateIdentitiesFor(walletAddress);
```
Comment (shannonwells) [inline src/core/contracts/registry.ts:242]:
```suggestion
  const identityAddresses: EthereumAddress[] = await getDelegateIdentitiesFor(walletAddress);
```

**Review by shannonwells — APPROVED** [review]
🚢

**Review by aneyzberg — APPROVED** [review]
🚢

**Pull request merged**